### PR TITLE
[New Profile] Add back-end validation for dates; cleanup

### PR DESCRIPTION
**Pull request by johnsaigle**
## Brief summary of changes

Complementary effort to #5758 -- dates are now validated on the back-end as well as the front-end.

Also cleans up some of the related code.

#### Testing instructions (if applicable)

1. Create a new candidate in `new_profile`. Try setting the date of birth to a date greater than today's date. You should get an error.

### CODE REVIEW

File: modules/new_profile/php/new_profile.class.inc [hunk 1]
```diff
@@ -1,62 +1,43 @@
...
 {
-    public  $skipTemplate = true;
+    public $skipTemplate = true;
     private $_candID;
     private $_pscid;
     private $_error;
 
     /**
-     * Tie the access to a data_entry permission
+     * Determines whether a user can access the page. The access to tied to
```
Comment (maltheism) [inline modules/new_profile/php/new_profile.class.inc:21]: Maybe "The access is tied to the data_entry permission." would be better?

**Comment by maltheism** [comment]
Everything looks good from reviewing the code. I'll try manually testing it.

**Comment by maltheism** [comment]
Hey @johnsaigle your backend PR throws an error successfully. I did encounter an issue but it has nothing to do with your PR and I'll be making an issue about it soon.

**Comment by johnsaigle** [comment]
Thanks for testing!

**Comment by johnsaigle** [comment]
Blocked by #6733

**Pull request merged**